lib/driver.rb

```diff
@@ -0,0 +1,83 @@
+require 'csv'
+require 'time'
+require_relative 'csv_record'
+require_relative 'trip_dispatcher'
+
+module RideShare
+  class Driver < CsvRecord
+    attr_reader :id, :name, :vin, :trips
+    attr_accessor :status
+
+    def initialize(id:, name:, vin:, status: :AVAILABLE, trips: nil)
+      super(id)
+      @name = name
+      @vin = vin #17 char
+      @status = status
+      @trips = trips || []
+      
+  
+      if status.class == String
+        @status = status.to_sym
+      end
+
+      if vin.class == Integer
+        vin.to_s
+      end
+
+      valid_statuses = [:AVAILABLE, :UNAVAILABLE]
+      raise ArgumentError if !valid_statuses.include?(@status)
+
+      raise ArgumentError if id <= 0 || id.nil?
+
+      raise ArgumentError if vin.length != 17
+    end 
+
+    def add_trip(trip)
+      trips << trip
+    end
+    
+    def average_rating
+      total_ratings = 0
+      @trips.each do |trip|
+        total_ratings += trip.rating
+      end
+      if total_ratings == 0
+        return 0
+      else
+        return average_rating = total_ratings.to_f / @trips.length
+      end
+    end
+
+    def total_revenue
+      total_revenue = 0
+      if @trips == nil
+        return 0
+      end 
+      @trips.each do |trip|
+        if trip.cost <= 1.65
+          total_revenue += 0
+        else
+          trip_pay = trip.cost - 1.65
+          trip_pay = trip_pay * 0.80
+          total_revenue += trip_pay
+        end
+      end
```

Comment:
Using `each` makes sense here.  The logic is kind of complicated.

In case you're curious about the `Enumerable` version:
```suggestion
      total_revenue = @trips.select { |trip| trip.cost > 1.65 }.sum { |trip| (trip.cost - 1.65) * 0.8 }
```